src/backend/gporca/libnaucrates/src/parser/CParseHandlerLogicalCTAS.cpp

```diff
@@ -232,11 +238,13 @@ CParseHandlerLogicalCTAS::EndElement(const XMLCh *const,  // element_uri,
 	IMdIdArray *distr_opfamilies =
 		dynamic_cast<CParseHandlerMetadataIdList *>(opfamilies_parse_handler)
 			->GetMdIdArray();
+	GPOS_ASSERT(NULL != distr_opfamilies);
 	distr_opfamilies->AddRef();
 
 	IMdIdArray *distr_opclasses =
 		dynamic_cast<CParseHandlerMetadataIdList *>(opclasses_parse_handler)
 			->GetMdIdArray();
```

Comment:
The dynamic_cast result is accessed on the same line. To check, you need to make the conversion into a separate variable. The same thing is above.
```suggestion
   CParseHandlerMetadataIdList* metadata = dynamic_cast<CParseHandlerMetadataIdList *>(opclasses_parse_handler);
   GPOS_ASSERT(NULL != metadata);
	IMdIdArray *distr_opclasses = metadata->GetMdIdArray();
```
By the way are casting is necessary here?

Reply:
fixed

Reply:
So why do we need type casting here? It looks like opclasses_parse_handler already has the CParseHandlerMetadataIdList type.

Reply:
indeed, fixed